pkg/kubectl/cmd/delete.go

```diff
@@ -321,8 +321,9 @@ func (o *DeleteOptions) deleteResource(info *resource.Info, deleteOptions *metav
 	// TODO: Remove this in or after 1.12 release.
 	//       Server version >= 1.11 no longer needs this hack.
 	mapping := info.ResourceMapping()
-	if mapping.Resource.GroupResource() == (schema.GroupResource{Group: "extensions", Resource: "daemonsets"}) ||
-		mapping.Resource.GroupResource() == (schema.GroupResource{Group: "apps", Resource: "daemonsets"}) {
+	if (mapping.Resource.GroupResource() == (schema.GroupResource{Group: "extensions", Resource: "daemonsets"}) ||
+		mapping.Resource.GroupResource() == (schema.GroupResource{Group: "apps", Resource: "daemonsets"})) &&
+		(*deleteOptions.PropagationPolicy == metav1.DeletePropagationOrphan) {
```

Comment:
shouldn't it be `!=` possibly with checking the nil value of delete options?

Reply:
It never will be nil, we're setting it to `metav1.DeletePropagationBackground` always a couple of lines before calling `deleteResource` and only conditionally setting `metav1.DeletePropagationOrphan`.

Reply:
ok, but the hack (forcing the pods to be deleted) should only be called if the policy **is not** DeletePropagationOrphan, right?

Reply:
/hold